File: src/dynamic_programming/cpp/climb_stairs/test/test_climb.cpp
#include "climb.h"
#include "gtest/gtest.h"

TEST(TestClimbBasic, TestWith0) {
  int num = 0;
  int result = 1;
  EXPECT_EQ(countWaysToClimbBasic(num), result);
}

TEST(TestClimbBasic, TestWith1) {
  int num = 1;
  int result = 1;
  EXPECT_EQ(countWaysToClimbBasic(num), result);
}

TEST(TestClimbBasic, TestWith2) {
  int num = 2;
  int result = 2;
  EXPECT_EQ(countWaysToClimbBasic(num), result);
}

TEST(TestClimbBasic, TestWith10) {
  int num = 10;
  int result = 274;
  EXPECT_EQ(countWaysToClimbBasic(num), result);
}

TEST(TestClimbBasic, TestWith20) {
  int num = 20;
  int result = 121415;
  EXPECT_EQ(countWaysToClimbBasic(num), result);
}

TEST(TestClimbMemo, TestWith0) {
  int num = 0;
  int result = 1;
  EXPECT_EQ(countWaysToClimbMemo(num), result);
}

TEST(TestClimbMemo, TestWith1) {
  int num = 1;
  int result = 1;
  EXPECT_EQ(countWaysToClimbMemo(num), result);
}

TEST(TestClimbMemo, TestWith2) {
  int num = 2;
  int result = 2;
  EXPECT_EQ(countWaysToClimbMemo(num), result);
}

TEST(TestClimbMemo, TestWith10) {
  int num = 10;
  int result = 274;
  EXPECT_EQ(countWaysToClimbMemo(num), result);
}

TEST(TestClimbMemo, TestWith20) {
  int num = 20;
  int result = 121415;
  EXPECT_EQ(countWaysToClimbMemo(num), result);
}

TEST(TestClimbTable, TestWith0) {
  int num = 0;
  int result = 1;
  EXPECT_EQ(countWaysToClimbTable(num), result);
}

TEST(TestClimbTable, TestWith1) {
  int num = 1;
  int result = 1;
  EXPECT_EQ(countWaysToClimbTable(num), result);
}

TEST(TestClimbTable, TestWith2) {
  int num = 2;
  int result = 2;
  EXPECT_EQ(countWaysToClimbTable(num), result);
}

TEST(TestClimbTable, TestWith10) {
  int num = 10;
  int result = 274;
  EXPECT_EQ(countWaysToClimbTable(num), result);
}

TEST(TestClimbTable, TestWith20) {
  int num = 20;
  int result = 121415;
  EXPECT_EQ(countWaysToClimbTable(num), result);
}

File: src/dynamic_programming/python/climbing_stairs/tests/test_climb.py
import unittest

from ..src.climb import (
    count_ways_to_climb_basic,
    count_ways_to_climb_memo,
    count_ways_to_climb_table,
)


class TestClimbBasic(unittest.TestCase):
    def test_with_0(self):
        num = 0
        result = 1
        self.assertEqual(count_ways_to_climb_basic(num), result)

    def test_with_1(self):
        num = 1
        result = 1
        self.assertEqual(count_ways_to_climb_basic(num), result)

    def test_with_2(self):
        num = 2
        result = 2
        self.assertEqual(count_ways_to_climb_basic(num), result)

    def test_with_10(self):
        num = 10
        result = 274
        self.assertEqual(count_ways_to_climb_basic(num), result)

    def test_with_20(self):
        num = 20
        result = 121415
        self.assertEqual(count_ways_to_climb_basic(num), result)


class TestClimbMemo(unittest.TestCase):
    def test_with_0(self):
        num = 0
        result = 1
        self.assertEqual(count_ways_to_climb_memo(num), result)

    def test_with_1(self):
        num = 1
        result = 1
        self.assertEqual(count_ways_to_climb_memo(num), result)

    def test_with_2(self):
        num = 2
        result = 2
        self.assertEqual(count_ways_to_climb_memo(num), result)

    def test_with_10(self):
        num = 10
        result = 274
        self.assertEqual(count_ways_to_climb_memo(num), result)

    def test_with_20(self):
        num = 20
        result = 121415
        self.assertEqual(count_ways_to_climb_memo(num), result)


class TestClimbTable(unittest.TestCase):
    def test_with_0(self):
        num = 0
        result = 1
        self.assertEqual(count_ways_to_climb_table(num), result)

    def test_with_1(self):
        num = 1
        result = 2
        self.assertEqual(count_ways_to_climb_table(num), result)

    def test_with_2(self):
        num = 2
        result = 2
        self.assertEqual(count_ways_to_climb_table(num), result)

    def test_with_10(self):
        num = 10
        result = 274
        self.assertEqual(count_ways_to_climb_table(num), result)

    def test_with_20(self):
        num = 20
        result = 121415
        self.assertEqual(count_ways_to_climb_table(num), result)


if __name__ == "__main__":
    unittest.main()
